spec/controllers/tag_wranglers_controller_spec.rb

```diff
@@ -79,6 +79,19 @@
                       [tag1.name, tag1.updated_at.to_s, tag1.type, "", "", "false"]])
           end
 
+          it "limits the number of tags" do
+            stub_const("ArchiveConfig", OpenStruct.new(ArchiveConfig))
+            ArchiveConfig.WRANGLING_REPORT_LIMIT = 1
+            tag1 = create(:tag, last_wrangler: user)
+            create(:tag, last_wrangler: user)
+
+            get :report_csv, params: { id: user.login }
+            result = CSV.parse(response.body.encode("utf-8")[1..], col_sep: "\t")
+
+            expect(result.length).to eq(2)
+            expect(result[1][0]).to eq(tag1.name)
```

Comment:
I was split on whether this should be the most recently wrangled (not indexed) or based on the tag's ID. With find_each, we get batching [but sacrifice the ability to order by something other than id](https://apidock.com/rails/ActiveRecord/Batches/find_each#:~:text=NOTE%3A%20It%E2%80%99s%20not%20possible%20to%20set%20the%20order.%20That%20is%20automatically%20set%20to%20ascending%20on%20the%20primary%20key%20(%E2%80%9Cid%20ASC%E2%80%9D)%20to%20make%20the%20batch%20ordering%20work.%20This%20also%20means%20that%20this%20method%20only%20works%20when%20the%20primary%20key%20is%20orderable%20(e.g.%20an%20integer%20or%20string).); I could change it to be descending though if that makes sense

Reply:
[Order for batch queries](https://api.rubyonrails.org/v6.1/classes/ActiveRecord/Batches.html#method-i-find_in_batches) is supported in Rails 6.1+, so we should be OK.

As @sarken mentioned elsewhere, the output should be the x most recently wrangled tags.

Reply:
Just FYI, using `updated_at` for the "most recently wrangled" check will start breaking very badly once https://github.com/otwcode/otwarchive/pull/4230 is in. You'll just end up getting a list of the most popular tags. `updated_at` isn't an accurate measure of when the `last_wrangler` last touched the tag (and [never has been](https://otwarchive.atlassian.net/browse/AO3-3631)).

Reply:
@redsummernight do you mean the `:order` option? That's only for the primary key, so neither `:asc` or `:desc` would correlate with wrangling time frame

Reply:
Ah, I missed that it's only on the primary key.

Reply:
`:desc` would, I think, have a better chance at showing *some* tags wrangled more recently (though not all), since it would show newer tags. But I've never been 100% clear on the purpose of this CSV — is it to check whether someone has been messing around with older tags that they shouldn't have been modifying, or to check whether someone has been doing their assigned duties of processing newer tags?